src/platform/unittest/TlsTest.cpp

```diff
@@ -932,4 +932,93 @@ TEST_P(TlsTest, PacketEncryptionPerf)
     }
 }
 
+uint64_t LockedCounter(
+    const uint64_t LoopCount
+    )
+{
+    uint64_t Start, End, Accumulator = 0;
+    QUIC_DISPATCH_LOCK Lock;
+    uint64_t Counter = 0;
+
+    QuicDispatchLockInitialize(&Lock);
+
+    for (uint64_t j = 0; j < LoopCount; ++j) {
+        Start = QuicTimeUs64();
```

Comment:
If you put the start/stop time in the loop you end up skewing the results I believe. Especially if things are really fast. Try putting it outside the loop and measure the time for the whole loop.

Reply:
But then you collect the time for initializing, comparing, and incrementing the counter, as well as the jump instruction.